### ISSUE TITLE
Unable to reproduce results with multiple runs

### ISSUE THREAD

**Issue by meghana-shastri24** [KaiyangZhou/deep-person-reid#485]
Results are random when trained multiple times with exactly the same configuration. Can you please help me understand why this is the case?

I am attaching the config for reference. I trained the GRID dataset on osnet_ain two times to verify this behavior. 

**First Run: Rank1 - 11.2 %, mAP 18.7 %
Second Run: Rank1 - 13.6 %, mAP 19.8 %**

Configuration
Show configuration
adam:
  beta1: 0.9
  beta2: 0.999
cuhk03:
  classic_split: False
  labeled_images: False
  use_metric_cuhk03: False
data:
  combineall: False
  height: 256
  k_tfm: 1
  load_train_targets: False
  norm_mean: [0.485, 0.456, 0.406]
  norm_std: [0.229, 0.224, 0.225]
  sources: ['grid']
  split_id: 0
  targets: ['grid']
  transforms: ['random_flip', 'color_jitter']
  type: image
  width: 128
  workers: 4
loss:
  name: softmax
  softmax:
    label_smooth: True
  triplet:
    margin: 0.3
    weight_t: 1.0
    weight_x: 0.0
market1501:
  use_500k_distractors: False
model:
  load_weights: 
  low_memory: False
  name: osnet_ain_x1_0
  pretrained: True
  resume: 
rmsprop:
  alpha: 0.99
sampler:
  num_cams: 1
  num_datasets: 1
  num_instances: 4
  train_sampler: RandomSampler
  train_sampler_t: RandomSampler
sgd:
  dampening: 0.0
  momentum: 0.9
  nesterov: False
test:
  batch_size: 300
  cross_camera: False
  dist_metric: cosine
  eval_freq: 1
  evaluate: False
  normalize_feature: False
  ranks: [1, 5, 10, 20]
  rerank: False
  start_eval: 0
  visrank: False
  visrank_topk: 10
train:
  base_lr_mult: 0.1
  batch_size: 64
  fixbase_epoch: 10
  gamma: 0.1
  lr: 0.0015
  lr_scheduler: cosine
  max_epoch: 100
  new_layers: ['classifier']
  open_layers: ['classifier']
  optim: amsgrad
  print_freq: 20
  seed: 1
  staged_lr: False
  start_epoch: 0
  stepsize: [20]
  weight_decay: 0.0005
use_gpu: True
video:
  pooling_method: avg
  sample_method: evenly
  seq_len: 15

Any help is appreciated...****
